File: judo-sdk/src/main/java/com/judopay/IdealWebViewClient.java
package com.judopay;

import android.graphics.Bitmap;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import java.net.MalformedURLException;
import java.net.URL;

public class IdealWebViewClient extends WebViewClient {
    private IdealWebViewCallback callback;
    private String merchantRedirectUrl;

    IdealWebViewClient(final IdealWebViewCallback callback, String merchantRedirectUrl) {
        this.callback = callback;
        this.merchantRedirectUrl = merchantRedirectUrl;
    }

    @Override
    public void onPageStarted(final WebView view, final String url, final Bitmap favicon) {
        super.onPageStarted(view, url, favicon);
        try {
            String host = new URL(url).getHost();
            String merchantRedirectHost = new URL(merchantRedirectUrl).getHost();
            if (host.contains(merchantRedirectHost)) {
                String checksum = url.split("cs=")[1];
                callback.onPageStarted(checksum);
            }
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
    }
}

File: judokit-android/src/main/java/com/judokit/android/ui/ideal/components/IdealWebViewClient.kt
package com.judokit.android.ui.ideal.components

import android.graphics.Bitmap
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.core.net.toUri

private const val CHECKSUM_DELIMITER = "cs="

class IdealWebViewClient(private val merchantRedirectUrl: String) :
    WebViewClient() {

    override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
        super.onPageStarted(view, url, favicon)
        val merchantRedirectHost = merchantRedirectUrl.toUri().host
        if (!url.isNullOrEmpty()) {
            val urlHost = url.toUri().host
            if (
                !merchantRedirectHost.isNullOrEmpty() &&
                !urlHost.isNullOrEmpty() &&
                urlHost.contains(merchantRedirectHost)
            ) {
                val urlSplit = url.split(CHECKSUM_DELIMITER)
                if (urlSplit.isNotEmpty() && urlSplit.size == 2) {
                    val checksum = urlSplit[1]
                    (view as IdealWebViewCallback).onPageStarted(checksum)
                }
            }
        }
    }
}
